### DISCUSSION TITLE
HDDM in HSSM —— Group and Individual Prameters

### DISCUSSION THREAD

**Discussion by JoeSu112** [lnccbrown/HSSM#345] [Q&A] [answered]
Hi all,
I am now trying to transition my data from HDDM to HSSM, and I have some problem on the interpretation of group- and subject-level parameters:

For example, in HDDM,  I can get the group mean parameter for threshold "a", and individual subject parameters "a_subj.i". The result table is as followed: 
|  | mean |
|--------|-----|
| a | ....|
| a_subj.1|....|
| a_subj.2|....|
| a_subj.3|....|

Now I want to the get group and individual parameters in HSSM as well, so I tried putting "a ~ 1 + (1|subject)" in the model. The result table is as followed:
|  | mean |
|--------|-----|
| a_Intercept | ....|
| a_1｜subject[1]|....|
| a_1｜subject[2]|....|
| a_1｜subject[3]|....|

And my questions are:
Is this a correct way to get group and individual parameters in HSSM? If it's wrong, what is the correct way to do so?
If it's correct, how should I interpret the result? Is the individual parameter "a_Intercept +  a_1｜subject[i]" or "a_1｜subject[i]" only or something else? Also, how about the group parameter?

Thanks for any help!!!

**Comment by JamesWeiChen** [comment]
@frankmj I have the same question~ thanks a lot!

**Comment by frankmj** [comment]
The recent release fixed a lot of the bugs that were popping up with
hierarchical estimation and Alex (cc'd) is in the process of making some
tutorials on how to estimate and interpret hierarchical models that will
make all this much clearer. But briefly, yes if you have an intercept then
the individual subject parameters are the random effects on top of that
fixed intercept so if you wanted that individual's total a you would add
the random effect to the intercept.

**Reply by JoeSu112** [reply]
Hi @frankmj ,

Thanks so much for your reply. Now I can calculate the individual subject parameters, but how about the group mean parameter? Can I just take the average of all the individual subject parameters? Or there is another way to calculate the group mean parameter?

Thanks again for your help!!

**Comment by frankmj** [comment, marked as answer]
In your setting the intercept is the group mean (fixed effect).

**Reply by zaizibai** [reply]
Hi @frankmj ,

thanks for making this wonderful package. It's nice that HSSM can allow the users to estimate the random slope in GLM function. But i have one question related to the individual parameter. If i estimate a model which allows drift rate to verify like the following way:
"v~ 1 + (1+congruence|participant) ". where congruence is a contrast coding variable.
To get the participant congruent or incongruent trials drift rate, i need to extract the samples of v_1｜participant[1], v_congrunece | participant[1] and then do the GLM calculation. Is this the correct way? 

Thanks a lot!

**Comment by frankmj** [comment]
I'm not sure what you mean by doing "the GLM calculation", as the model is
already estimating the GLM in a hierarchical bayesian way. But you are
right that if you want to extract each participant's effect of congruence
you would look at the posteriors for those individual coefficients. If you
want to know whether they are consistently different than zero across the
population then you would not perform a frequentist analysis of these
individual coefficients but instead you would look at the group level
distribution for v_congruence and see whether the HDI overlaps with 0 or
not.

Michael

Michael J Frank, PhD | Edgar L. Marston Professor
Director, Carney Center for Computational Brain Science
<https://www.brown.edu/carney/ccbs>
Laboratory of Neural Computation and Cognition <https://www.lnccbrown.com/>

Brown University
website <http://ski.clps.brown.edu>
